### PR TITLE
check netid on start

### PR DESCRIPTION
**Pull request by mattkasun**
- compare netID on during initialization and update netID/DefautlInterface if contains uppercase or a . 
- setDefaults for networks

### CODE REVIEW

**Comment by mattkasun** [comment]
fixes #1611

**Review by theguy951357 — APPROVED** [review]
good

File: main.go [hunk 1]
```diff
@@ -74,6 +75,38 @@ func initialize() { // Client Mode Prereq Check
 	if err = database.InitializeDatabase(); err != nil {
 		logger.FatalLog("Error connecting to database")
 	}
+	//update network defaults
+	networks, _ := logic.GetNetworks()
```
Comment (0xdcarns) [inline main.go:79]: this whole block should probably be in another function

Comment (0xdcarns) [reply main.go:79]: further, you could add this logic to setNetworkDefaults in `serverctl.go:124`

File: serverctl/serverctl.go [hunk 2]
```diff
@@ -132,9 +132,32 @@ func setNetworkDefaults() error {
...
+			newNet.SetDefaults()
+			if err := logic.SaveNetwork(&newNet); err != nil {
+				logger.Log(0, "error saving networks during initial update:", err.Error())
+			}
+			if err := logic.DeleteNetwork(net.NetID); err != nil {
+				logger.Log(0, "error deleting old network:", err.Error())
+			}
+		} else {
+			net.SetDefaults()
+			_, _, _, _, _, _, err = logic.UpdateNetwork(&net, &net)
```
Comment (0xdcarns) [inline serverctl/serverctl.go:157]: should this not just be logic.SaveNetwork? since you're updating with the same network struct?

Comment (mattkasun) [reply serverctl/serverctl.go:157]: Update network calls validation

Comment (0xdcarns) [reply serverctl/serverctl.go:157]: ah okay, probably need to add that to save in the future then

**Review by 0xdcarns — APPROVED** [review]

**Review by afeiszli — APPROVED** [review]

**Pull request merged**